### PR TITLE
changed lasfilter for lasreader fast aoi in rlas package

### PR DESCRIPTION
**Pull request by floriandeboissieu**
Changed lasfilter queries for lasreader fast AOI queries (see rlas package), which makes the extraction of rectangle/circle faster with indexed las/laz files (see [link](https://rapidlasso.com/2012/12/03/lasindex-spatial-indexing-of-lidar-data/)).

### CODE REVIEW

**Comment by floriandeboissieu** [comment]
sorry, I pull-request on the master branch... should I make it again on devel branch?

**Comment by Jean-Romain** [comment]
I prefer the devel branch, but currently the two branches are even so it not a problem. I'm waiting to resolve your PR in rlas before to study that one.

File: R/catalog_query.r [hunk 1]
```diff
@@ -35,7 +35,8 @@
 #' It works only for tiles that are arranged in gridlines.
 #'
 #' @aliases catalog_queries
-#' @param obj A Catalog object
+#' @param obj A Catalog object        if(!lasfilter.filter(&lasreader->point))
```
Comment (Jean-Romain) [inline R/catalog_query.r]: ?? wrong ctrl-C ctrl-V

Comment (floriandeboissieu) [inline R/catalog_query.r]: Exact, sorry...

**Comment by Jean-Romain** [comment]
I'm curious, did you benchmarked the computation time with and without LAX file?

**Comment by Jean-Romain** [comment]
I merge manually but I did something wrong when correcting your error. Anyway I merged properly in `rlas` it's the most important.

**Pull request merged**